### PR TITLE
enable randomness in genesis

### PR DESCRIPTION
**Pull request by zjma**
## Description

Enable randomness by default, but disable it for a few tests which may alter validator set, or expect immediate reconfigs.

### CODE REVIEW

File: testsuite/smoke-test/src/consensus/dag/dag_fault_tolerance.rs [hunk 1]
```diff
@@ -67,17 +67,12 @@ async fn test_no_failures() {
...
             assert!(
                 executed_transactions >= 4,
                 "no progress with active consensus, only {} transactions",
                 executed_transactions
             );
-            assert!(
-                executed_rounds >= 2,
-                "no progress with active consensus, only {} rounds",
-                executed_rounds
-            );
```
Comment (zjma) [inline testsuite/smoke-test/src/consensus/dag/dag_fault_tolerance.rs:80]: isn't round number reset every epoch? why this assert?
@ibalajiarun @zekun000

Comment (zekun000) [reply testsuite/smoke-test/src/consensus/dag/dag_fault_tolerance.rs:80]: i thought the executed rounds are the difference between start and end round?

**Review by danielxiangzl — APPROVED** [review]

File: testsuite/smoke-test/src/consensus/dag/dag_fault_tolerance.rs (hunk 1 above)
Comment (zjma) [reply testsuite/smoke-test/src/consensus/dag/dag_fault_tolerance.rs:80]: with randomness enabled this is what happened in the test.
```
cycle 0 lasted 5.011 with 1 epochs, 3 rounds and 16 transactions
All at epochs: [2, 2, 2], from [1, 1, 1]
All at rounds: [6, 6, 6], from [3, 3, 3]
All at versions: [20, 20, 20], from [4, 4, 4]
cycle 0 lasted 5.010 with 1 epochs, 0 rounds and 8 transactions
All at epochs: [2, 2, 2], from [1, 1, 1]
All at rounds: [1, 1, 1], from [1, 1, 1]
All at versions: [10, 10, 10], from [2, 2, 2]
```

Comment (zjma) [reply testsuite/smoke-test/src/consensus/dag/dag_fault_tolerance.rs:80]: without randomness:
```
cycle 0 lasted 5.011 with 1 epochs, 3 rounds and 16 transactions
All at epochs: [2, 2, 2], from [1, 1, 1]
All at rounds: [6, 6, 6], from [3, 3, 3]
All at versions: [20, 20, 20], from [4, 4, 4]
```

Comment (zjma) [reply testsuite/smoke-test/src/consensus/dag/dag_fault_tolerance.rs:80]: i believe today it's passing the round assert just by luck?

**Review by zekun000 — APPROVED** [review]

**Pull request merged**